core/src/Kokkos_View.hpp

```diff
@@ -79,9 +79,10 @@ class ViewMapping {
 };
 
 template <typename IntType>
-KOKKOS_INLINE_FUNCTION std::size_t count_valid_integers(
-    const IntType i0, const IntType i1, const IntType i2, const IntType i3,
-    const IntType i4, const IntType i5, const IntType i6, const IntType i7) {
+inline std::size_t count_valid_integers(const IntType i0, const IntType i1,
```

Comment:
We're calling this from a host function only thus the annotation creates a compiler warning.

Reply:
Calling a `__host__ __device__` function from a `__host__` function should be fine.  I expect the inverse is what would yield a warning.

Reply:
Does that make sense? I would only expect a warning if we call a host-only function from a `host`-`device` function but not if we call a `host-device` function from a host function.

Reply:
My comment was imprecise: We're get the compiler warning from using std::string in a `__host__` `__device__` function.

Reply:
Ah, but that only applies to `runtime_check_rank`, right? Anyway, I think it's fine since that is the only function to call this one anyway.

Reply:
Yea, thanks Daniel for looking at that. The troublesome issue that I am having right now is that some call path from TestViewLayoutStrideAssignment.hpp:638 is giving `Constructor for Kokkos View 'LayoutStride_mirror' has mismatched number of arguments. Number of arguments = 8 but dynamic rank = 1` **if** we always check the parameters, that is, if we remove that conditional in the allocating View ctor. I am not sure if this is wanted or if we are hiding an issue which also did not exhibit in the previous implementation. There is likely an issue because CI shows errors and also this reported issue: https://github.com/kokkos/kokkos/issues/2995
This seems to be related.
More details to the CI: failing test:
```
[ RUN ] cuda_DeathTest.view_layoutstride_right_to_layoutleft_assignment
view_layoutstride_right_to_layoutleft_assignment: srand(1639811970)
Constructor for Kokkos View 'LayoutStride_mirror' has mismatched number of arguments. Number of arguments = 8 but dynamic rank = 1
```